libcontainer/specconv/spec_linux.go

```diff
@@ -650,13 +654,29 @@ func stringToDeviceRune(s string) (configs.DeviceType, error) {
 	}
 }
 
-func createDevices(spec *specs.Spec, config *configs.Config) error {
-	// Add default set of devices.
-	for _, device := range AllowedDevices {
-		if device.Path != "" {
-			config.Devices = append(config.Devices, device)
+func createDevices(spec *specs.Spec, config *configs.Config) ([]*configs.Device, error) {
+
+	// If a spec device is redundant with a default device, remove that default
+	// device (the spec one takes priority).
+	dedupedAllowDevs := []*configs.Device{}
+
+	if spec.Linux != nil {
+		for _, ad := range AllowedDevices {
+			conflict := false
+			for _, sd := range spec.Linux.Devices {
+				if sd.Path == ad.Path {
+					conflict = true
```

Comment:
Maybe add `break` here, since it does not make sense to keep iterating over `spec.Linux.Devices` once you found a conflict?

Reply:
Good idea, thanks.